maint/config_filter.pl

```diff
@@ -0,0 +1,332 @@
+    $ctype_hash{COMPLEX16} = "double _Complex";
+    $ctype_hash{COMPLEX32} = "long double _Complex";
+}
+if($have_hash{CXX_BINDING}){
+    if($size_hash{C_BOOL}>0){
+        if($size_hash{CXX_BOOL}==$size_hash{C_BOOL}){
+            $ctype_hash{CXX_BOOL}="_Bool";
+        }
+        else{
+            $ctype_hash{CXX_BOOL} = "unsigned ".$sized_types{"i".$size_hash{CXX_BOOL}};
+        }
+    }
+    if($size_hash{C_FLOAT_COMPLEX}>0 && $size_hash{C_FLOAT_COMPLEX}==$size_hash{CXX_FLOAT_COMPLEX}){
+        $ctype_hash{CXX_FLOAT_COMPLEX} = "float _Complex";
+    }
+    if($size_hash{C_DOUBLE_COMPLEX}>0 && $size_hash{C_DOUBLE_COMPLEX}==$size_hash{CXX_DOUBLE_COMPLEX}){
+        $ctype_hash{CXX_DOUBLE_COMPLEX} = "double _Complex";
+    }
+    if($size_hash{C_LONG_DOUBLE_COMPLEX}>0 && $size_hash{C_LONG_DOUBLE_COMPLEX}==$size_hash{CXX_LONG_DOUBLE_COMPLEX}){
+        $ctype_hash{CXX_LONG_DOUBLE_COMPLEX} = "long double _Complex";
+    }
+}
+my @lines;
+open In, "src/include/mpi.h" or die "Can't open src/include/mpi.h.\n";
+while(<In>){
+    if(/^#define\s+MPIX?_(\w+)\s*\(\(MPI_Datatype\)(0x[0-9a-f]+)\)/){
+        my ($name, $id) = ($1, $2);
+        if($name ne "DATATYPE_NULL"){
+            my $t_name=$name;
+            $t_name=~s/(UNSIGNED|SIGNED)_//;
+            my $size = $size_hash{$t_name};
+            if(!$size and $name=~/^2(\w+)/){
+                $size = 2 * $size_hash{$1};
+                $size_hash{$name}=$size;
+            }
+            if($id=~/0x8c/ and $size ne "0"){
+                $size_hash{$name} = "-";
+            }
+            elsif($size eq "-"){
+            }
+            elsif(!$size){
+            }
+            else{
+                substr($id,6,2)=sprintf("%02x", $size);
+                s/0x[0-9a-f]+/$id/;
+            }
+            $MPI_hash{$name} = $id;
+        }
+    }
+    elsif(/^#define\s+MPIX?_(\w+)\s+MPIX?_(\w+)/){
+        $MPI_hash{$1} = $MPI_hash{$2};
+        $size_hash{$1} = $size_hash{$2};
+    }
+    elsif(/PLACEHOLDER FOR config_filter/){
+        my $type = $ctype_hash{FINT};
+        push @lines, "typedef ".$ctype_hash{FINT}." MPI_Fint;\n";
+        my $type = $ctype_hash{AINT};
+        push @lines, "typedef ".$ctype_hash{AINT}." MPI_Aint;\n";
+        my $type = $ctype_hash{OFFSET};
+        push @lines, "typedef ".$ctype_hash{OFFSET}." MPI_Offset;\n";
+        my $type = $ctype_hash{COUNT};
+        push @lines, "typedef ".$ctype_hash{COUNT}." MPI_Count;\n";
+        push @lines, "\n";
+        next;
+    }
+    elsif(/^#ifdef\s+(HAVE_MPIX?_(\w+))/){
+        my ($have, $name) = ($1, $2);
+        my $t = 0;
+        if($size_hash{$name}){
+            $t = 1;
+        }
+        $_="#if $t /\x2a $have \x2a/\n";
+    }
+    push @lines, $_;
+}
+close In;
+open Out, ">src/include/mpi.h" or die "Can't write src/include/mpi.h.\n";
+print Out @lines;
+close Out;
+open Out, ">src/include/mpichconf.h" or die "Can't write src/include/mpichconf.h.\n";
+foreach my $l (@mpichconf_lines){
```

Comment:
Modifying `src/include/mpichconf.h` is a convenience. A better solution may be to create a separate header file and included in `mpi_ext.h` (as ROMIO probably need it too).